src/dmd/mars.d

```diff
@@ -1532,6 +1532,24 @@ private bool parseCommandLine(const ref Strings arguments, const size_t argc, re
         return (overflow || value > max || *p) ? uint.max : value;
     }
 
+    /********************************
+     * Params:
+     *  p = 0 terminated string
+     *  s = string
+     * Returns:
+     *  true if `p` starts with `s`
+     */
+    static pure bool startsWith(const(char)* p, string s)
+    {
+        foreach (const c; s)
+        {
+            if (c != *p)
+                return false;
+            ++p;
+        }
+        return true;
+    }
```

Comment:
I'm note sure if this is a problem, but since you're looping through `s` and not `p` and if `s` is longer than `p` and ends with `\0` it would continue the loop beyond `p`. Example:

```d
import std.stdio : writeln;

static bool startsWith(const(char)* p, string s)
{
    writeln(s.length);
    foreach (i, const c; s)
    {
        writeln(*p);
        if (c != *p)
            return false;
        ++p;
    }
    return true;
}

void main()
{
    auto r = startsWith("foo", "foo\0\0\0\0\0\0\0\0\0\0\0\0\0\0");
    writeln(r);
}
```

The above code prints the following on my machine:

```
17
f
o
o













`
false
```

Reply:
It's not an issue because none of the strings passed to `s` are user-defined, they're all string literals.

Reply:
@WalterBright  When can we stop using `char*` for strings and start using D's string facilities?

Reply:
It still feels wrong to have a broken function. What if someone copies it and starts to use it somewhere else with user-defined strings.